### PR TITLE
TPC: Update CheckOfSlices

### PR DESCRIPTION
**Pull request by mlesch**
Update:
-  Takes uncertainties of single data points into account for calculation of mean and checks
- Allows for Range and Zero Check
- Saving MetaData for QualityObserver

### CODE REVIEW

File: Modules/TPC/include/TPC/CheckOfSlices.h [hunk 1]
```diff
@@ -44,17 +44,33 @@ class CheckOfSlices : public o2::quality_control::checker::CheckInterface
   std::string getAcceptedType() override { return "TCanvas"; }
 
  private:
-  ClassDefOverride(CheckOfSlices, 2);
+  ClassDefOverride(CheckOfSlices, 3);
+  void calculateStatistics(const double* yValues, const double* yErrors, bool useErrors, const int firstPoint, const int lastPoint, double& mean, double& stddevOfMean);
```
Comment (wiechula) [inline Modules/TPC/include/TPC/CheckOfSlices.h:48]: I think this is copy'n'paste of `CheckOfTrendings::calculateStatistics`
Please put it in one common helper file e.g. `include/TPC/Utility.h`

File: Modules/TPC/src/CheckOfSlices.cxx [hunk 2]
```diff
@@ -110,184 +119,438 @@ Quality CheckOfSlices::check(std::map<std::string, std::shared_ptr<MonitorObject
...
+    checkMessage = mMediumString;
     msg->SetFillColor(kOrange);
   } else if (checkResult == Quality::Null) {
     h->SetFillColor(0);
+    msg->AddText("Quality::Null. Failed checks:");
+    checkMessage = mNullString;
+  }
+
+  // Split lines by hand as \n does not work with TPaveText
+  std::string delimiter = "\n";
```
Comment (wiechula) [inline Modules/TPC/src/CheckOfSlices.cxx:322]:
```suggestion
  const std::string delimiter = "\n";
```

**Review by wiechula — CHANGES_REQUESTED** [review]

File: Modules/TPC/include/TPC/CheckOfSlices.h (hunk 1 above)
Comment (mlesch) [reply Modules/TPC/include/TPC/CheckOfSlices.h]: I moved `calculateStatistics` to` include/TPC/Utility.h` and removed it from both `CheckOfTrendings` and `CheckOfSlices`

File: Modules/TPC/src/CheckOfSlices.cxx (hunk 2 above)
Comment (mlesch) [reply Modules/TPC/src/CheckOfSlices.cxx]: `Done, I also did a similar change now in CheckOfTrendings`

**Review by wiechula — CHANGES_REQUESTED** [review]

**Review by wiechula — APPROVED** [review]

**Pull request merged**